libvast/vast/aliases.hpp

```diff
@@ -61,5 +64,7 @@ constexpr id max_events = max_id + 1;
 /// The data type for an enumeration.
 using enumeration = uint32_t;
 
+using synopsis_options = caf::config_value_map;
```

Comment:
Thanks for following my proposed change. 👍

However, I think this aliases the wrong type. The `config_value_map` really is a `map<map<config_value>>` (I know the name is misleading, hence why I started the discussion over here: https://github.com/actor-framework/actor-framework/issues/802). This means any call to `put(synopsis_options_, key, std::move(value));` will fail, unless `key` has the format `<component>.<parameter-name>`.

Looking at the code, this seems not the case. For example, `meta_idx.set_synopsis_option("max-partition-size", caf::config_value{max_partition_size});` would result in nothing being added to the options map (missing category). Currently, `put` silently fails here (maybe not the best of ideas and `put` should instead automagically add a `global.` prefix). In any case, I think we don't want grouping by component/category.

To cut a long story short: I think this should read `using synopsis_options = caf::dictionary<value_map>;` instead and this PR lacks a unit test for this new functionality.

Reply:
That makes almost sense, I'm just puzzled by the type `value_map`. Did you mean `dictionary<config_value>`?

Reply:
Okay, the last two commits contain that change (plus a new unit test).